eclipse-collections/src/main/java/org/eclipse/collections/impl/set/immutable/AbstractImmutableSet.java

```diff
@@ -474,4 +474,37 @@ public boolean isProperSubsetOf(SetIterable<? extends T> candidateSuperset)
     {
         return this.toSet().asParallel(executorService, batchSize);
     }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (o == this)
+        {
+            return true;
+        }
+
+        if (!(o instanceof Set))
+        {
+            return false;
+        }
+
+        Set<?> s = (Set<?>) o;
+        if (s.size() != this.size())
+        {
+            return false;
+        }
+
+        try
+        {
+            return this.containsAll(s);
+        }
+        catch (ClassCastException ignored)
+        {
+            return false;
+        }
+        catch (NullPointerException ignored)
+        {
+            return false;
+        }
+    }
```

Comment:
Similarly out of place

Reply:
@motlin ok, and anyway findbugs was complaining `hashCode` to be missing, I moved it in the one subclass that delegated to super and that also implemented `hashCode`.